service/telemetry/otelconftelemetry/logger.go

```diff
@@ -71,11 +70,6 @@ func newLogger(set telemetry.Settings, cfg Config, sdk *config.SDK, res *resourc
 		}))
 	}
 
-	var lp log.LoggerProvider
-	if sdk != nil {
-		lp = sdk.LoggerProvider()
-	} else {
-		lp = noop.NewLoggerProvider()
-	}
+	lp := sdk.LoggerProvider()
```

Comment:
SDK creation is encapsulated in createProviders, which is now the only caller of newLogger. It is impossible for the SDK to be nil at this point.